desc/geometry/curve.py

```diff
@@ -637,29 +657,34 @@ def __init__(
     ):
         super().__init__(name)
         X, Y, Z = np.atleast_1d(X), np.atleast_1d(Y), np.atleast_1d(Z)
+        X, Y, Z = np.broadcast_arrays(X, Y, Z)
```

Comment:
what does this call accomplish?

Reply:
The main use case is if one of the coordinates is constant, ie X=[1,2,3...], Y=[4,5,6...], Z=0. We broadcast everything together to make sure they're the same length